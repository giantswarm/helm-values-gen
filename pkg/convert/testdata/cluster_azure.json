{
    "$schema": "https://json-schema.org/draft/2019-09/schema",
    "description": "Configuration of an Azure cluster using Cluster API",
    "properties": {
        "attachCapzControllerIdentity": {
            "title": "Attach controller identity",
            "type": "boolean",
            "default": false
        },
        "azure": {
            "properties": {
                "azureClusterIdentity": {
                    "description": "AzureClusterIdentity resource to use for this cluster.",
                    "properties": {
                        "name": {
                            "title": "Name",
                            "type": "string",
                            "default": "cluster-identity"
                        },
                        "namespace": {
                            "title": "Namespace",
                            "type": "string",
                            "default": "org-giantswarm"
                        }
                    },
                    "title": "Identity",
                    "type": "object"
                },
                "location": {
                    "title": "Location",
                    "type": "string",
                    "default": "westeurope"
                },
                "subscriptionId": {
                    "title": "Subscription ID",
                    "type": "string",
                    "default": "PLACEHOLDER"
                }
            },
            "title": "Azure settings",
            "type": "object"
        },
        "clusterDescription": {
            "description": "User-friendly description of the cluster's purpose.",
            "title": "Cluster description",
            "type": "string",
            "default": "test"

        },
        "clusterName": {
            "description": "Unique identifier, cannot be changed after creation.",
            "title": "Cluster name",
            "type": "string",
            "default": ""

        },
        "controlPlane": {
            "properties": {
                "etcdVolumeSizeGB": {
                    "title": "Etcd volume size (GB)",
                    "type": "integer",
                    "default": 10
                },
                "instanceType": {
                    "title": "Node VM size",
                    "type": "string",
                    "default": "Standard_D4s_v3"
                },
                "replicas": {
                    "title": "Number of nodes",
                    "type": "integer",
                    "default": 3
                },
                "rootVolumeSizeGB": {
                    "title": "Root volume size (GB)",
                    "type": "integer",
                    "default": 50
                }
            },
            "title": "Control plane settings",
            "type": "object"
        },
        "defaults": {
            "properties": {
                "evictionMinimumReclaim": {
                    "title": "Default settings for eviction minimum reclaim",
                    "type": "string",
                    "default": "imagefs.available=5%,memory.available=100Mi,nodefs.available=5%"
                },
                "hardEvictionThresholds": {
                    "title": "Default settings for hard eviction thresholds",
                    "type": "string",
                    "default": "memory.available<200Mi,nodefs.available<10%,nodefs.inodesFree<3%,imagefs.available<10%,pid.available<20%"
                },
                "softEvictionGracePeriod": {
                    "title": "Default settings for soft eviction grace period",
                    "type": "string",
                    "default": "memory.available=30s,nodefs.available=2m,nodefs.inodesFree=1m,imagefs.available=2m,pid.available=1m"
                },
                "softEvictionThresholds": {
                    "title": "Default settings for soft eviction thresholds",
                    "type": "string",
                    "default": "memory.available<500Mi,nodefs.available<15%,nodefs.inodesFree<5%,imagefs.available<15%,pid.available<30%"
                }
            },
            "title": "Default settings",
            "type": "object"
        },
        "enablePerClusterIdentity": {
            "title": "Enable identity per cluster",
            "type": "boolean",
            "default": false
        },
        "includeClusterResourceSet": {
            "description": "If true, a ClusterResourceSet resource is generated for the cluster.",
            "title": "Include ClusterResourceSet",
            "type": "boolean",
            "default": true
        },
        "kubernetesVersion": {
            "title": "Kubernetes version",
            "type": "string",
            "default": "1.24.8"
        },
        "machineDeployments": {
            "default": [
                {
                    "name": "md00",
                    "instanceType": "Standard_D2s_v3",
                    "replicas": 3,
                    "rootVolumeSizeGB": 50,
                    "disableHealthCheck": false,
                    "customNodeLabels": [],
                    "customNodeTaints": []
                }
            ],
            "items": {
                "properties": {
                    "customNodeLabels": {
                        "items": {
                            "title": "Label",
                            "type": "string"
                        },
                        "title": "Custom node labels",
                        "type": "array"
                    },
                    "customNodeTaints": {
                        "descriptions": "Taints that will be set on all nodes in the node pool, to avoid the scheduling of certain workloads.",
                        "items": {
                            "properties": {
                                "effect": {
                                    "enum": [
                                        "NoSchedule",
                                        "PreferNoSchedule",
                                        "NoExecute"
                                    ],
                                    "title": "Effect",
                                    "type": "string"
                                },
                                "key": {
                                    "title": "Key",
                                    "type": "string"
                                },
                                "value": {
                                    "title": "Value",
                                    "type": "string"
                                }
                            },
                            "required": [
                                "effect",
                                "key",
                                "value"
                            ],
                            "title": "Node taint",
                            "type": "object"
                        },
                        "title": "Custom node taints",
                        "type": "array"
                    },
                    "disableHealthCheck": {
                        "title": "Disable HealthChecks for the MachineDeployment",
                        "type": "boolean"
                    },
                    "instanceType": {
                        "title": "VM size",
                        "type": "string"
                    },
                    "name": {
                        "description": "Unique identifier, cannot be changed after creation.",
                        "title": "Name",
                        "type": "string"
                    },
                    "replicas": {
                        "title": "Number of nodes in the MachineDeployment",
                        "type": "integer"
                    },
                    "rootVolumeSizeGB": {
                        "title": "Root volume size (GB)",
                        "type": "integer"
                    }
                },
                "title": "Node pool",
                "type": "object"
            },
            "title": "Node pools",
            "type": "array"
        },
        "machinePools": {
            "$comment": "/machineDeployments is used instead.",
            "default": [],
            "deprecated": true,
            "items": {
                "properties": {
                    "customNodeLabels": {
                        "items": {
                            "title": "Label",
                            "type": "string"
                        },
                        "title": "Custom node labels",
                        "type": "array"
                    },
                    "customNodeTaints": {
                        "descriptions": "Taints that will be set on all nodes in the node pool, to avoid the scheduling of certain workloads.",
                        "items": {
                            "properties": {
                                "effect": {
                                    "enum": [
                                        "NoSchedule",
                                        "PreferNoSchedule",
                                        "NoExecute"
                                    ],
                                    "title": "Effect",
                                    "type": "string"
                                },
                                "key": {
                                    "title": "Key",
                                    "type": "string"
                                },
                                "value": {
                                    "title": "Value",
                                    "type": "string"
                                }
                            },
                            "required": [
                                "effect",
                                "key",
                                "value"
                            ],
                            "title": "Node taint",
                            "type": "object"
                        },
                        "title": "Custom node taints",
                        "type": "array"
                    },
                    "instanceType": {
                        "title": "VM size",
                        "type": "string"
                    },
                    "location": {
                        "title": "Location",
                        "type": "string"
                    },
                    "maxSize": {
                        "title": "Maximum number of nodes",
                        "type": "integer"
                    },
                    "minSize": {
                        "title": "Minimum number of nodes",
                        "type": "integer"
                    },
                    "name": {
                        "description": "Unique identifier, cannot be changed after creation.",
                        "title": "Name",
                        "type": "string"
                    },
                    "rootVolumeSizeGB": {
                        "title": "Root volume size (GB)",
                        "type": "integer"
                    }
                },
                "title": "Node pool",
                "type": "object"
            },
            "title": "Machine pools (deprecated)",
            "type": "array"
        },
        "network": {
            "properties": {
                "hostCIDR": {
                    "title": "Host CIDR",
                    "type": "string",
                    "default": "10.0.0.0/8"
                },
                "podCIDR": {
                    "title": "Pod CIDR",
                    "type": "string",
                    "default": "192.168.0.0/16"
                },
                "serviceCIDR": {
                    "title": "Service CIDR",
                    "type": "string",
                    "default": "172.31.0.0/16"
                }
            },
            "title": "Network settings",
            "type": "object"
        },
        "oidc": {
            "properties": {
                "caPem": {
                    "description": "Identity provider's CA certificate in PEM format.",
                    "title": "Certificate authority",
                    "type": "string",
                    "default": ""
                },
                "clientId": {
                    "title": "Client ID",
                    "type": "string",
                    "default": ""
                },
                "groupsClaim": {
                    "title": "Groups claim",
                    "type": "string",
                    "default": ""
                },
                "issuerUrl": {
                    "title": "Issuer URL",
                    "type": "string",
                    "default": ""
                },
                "usernameClaim": {
                    "title": "Username claim",
                    "type": "string",
                    "default": ""
                }
            },
            "title": "OIDC settings",
            "type": "object"
        },
        "organization": {
            "title": "Organization",
            "type": "string",
            "default": ""
        },
        "sshSSOPublicKey": {
            "title": "SSH Public key for SSO",
            "type": "string",
            "default": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDnfhWCBrplizz7XzX1Sc+I57TvCe+EEARTeYBuqWa3BRNVysSdPdbxHHJ6qzVY66snO4wq4yokno0yBdA7hAV73Y38/yZEzdKux+UyRwrnoerOdOw/590U6Pkm1mB9YhUQ4XrS5j67ZNUc/LeSQ+aWf65HATcLUccfIIVNt7OOEl+VGvg9Pvoukg+A/mSoYdBQssGNbzB+8niK5ohIsKMxS0aqaLlEZMtXxZuD8YDvIRO2m8n+djQ6wqi4HTVcBKRREa5blVu3Ug42O4XDvcdHxqHyPCUSU7ho/PJDR1B45gub+hKt/O4EdtzVZE4Z7hGQEfXUd5C0mbg+V5gcxLtt"
        }
    },
    "title": "Cluster configuration",
    "type": "object"
}
