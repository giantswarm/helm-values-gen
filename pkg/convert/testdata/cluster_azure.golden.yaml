attachCapzControllerIdentity: false
azure:
  azureClusterIdentity:
    name: cluster-identity
    namespace: org-giantswarm
  location: westeurope
  subscriptionId: PLACEHOLDER
clusterDescription: test
clusterName: ""
controlPlane:
  etcdVolumeSizeGB: 10
  instanceType: Standard_D4s_v3
  replicas: 3
  rootVolumeSizeGB: 50
defaults:
  evictionMinimumReclaim: imagefs.available=5%,memory.available=100Mi,nodefs.available=5%
  hardEvictionThresholds: memory.available<200Mi,nodefs.available<10%,nodefs.inodesFree<3%,imagefs.available<10%,pid.available<20%
  softEvictionGracePeriod: memory.available=30s,nodefs.available=2m,nodefs.inodesFree=1m,imagefs.available=2m,pid.available=1m
  softEvictionThresholds: memory.available<500Mi,nodefs.available<15%,nodefs.inodesFree<5%,imagefs.available<15%,pid.available<30%
enablePerClusterIdentity: false
includeClusterResourceSet: true
kubernetesVersion: 1.24.8
machineDeployments:
- customNodeLabels: []
  customNodeTaints: []
  disableHealthCheck: false
  instanceType: Standard_D2s_v3
  name: md00
  replicas: 3
  rootVolumeSizeGB: 50
machinePools: []
network:
  hostCIDR: 10.0.0.0/8
  podCIDR: 192.168.0.0/16
  serviceCIDR: 172.31.0.0/16
oidc:
  caPem: ""
  clientId: ""
  groupsClaim: ""
  issuerUrl: ""
  usernameClaim: ""
organization: ""
sshSSOPublicKey: ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDnfhWCBrplizz7XzX1Sc+I57TvCe+EEARTeYBuqWa3BRNVysSdPdbxHHJ6qzVY66snO4wq4yokno0yBdA7hAV73Y38/yZEzdKux+UyRwrnoerOdOw/590U6Pkm1mB9YhUQ4XrS5j67ZNUc/LeSQ+aWf65HATcLUccfIIVNt7OOEl+VGvg9Pvoukg+A/mSoYdBQssGNbzB+8niK5ohIsKMxS0aqaLlEZMtXxZuD8YDvIRO2m8n+djQ6wqi4HTVcBKRREa5blVu3Ug42O4XDvcdHxqHyPCUSU7ho/PJDR1B45gub+hKt/O4EdtzVZE4Z7hGQEfXUd5C0mbg+V5gcxLtt
